modules/collection/programs/zed.nix

```
@@ -0,0 +1,167 @@
+          formatter = {
+            external = {
+              command = "prettier";
+              arguments = ["--stdin-filepath" "{buffer_path}"];
+            };
+          };
+          tab_size = 2;
+        };
+        load_direnv = "shell_hook";
+        theme = {
+          mode = "system";
+          light = "One Light";
+          dark = "One Dark";
+        };
+      };
+      description = ''
+        Configuration written to {file}`$HOME/.config/zed/settings.json`.
+        Please reference [zed's documentation] for config options.
+
+        [zed's documentation]: https://zed.dev/docs/configuring-zed
+      '';
+    };
+
+    keymap = mkOption {
+      type = json.type;
+      default = {};
+      example = [
+        {
+          bindings = {
+            ctrl-right = "editor::SelectLargerSyntaxNode";
+            ctrl-left = "editor::SelectSmallerSyntaxNode";
+          };
+        }
+        {
+          bindings = {
+            o = "project_panel::Open";
+          };
+          context = "ProjectPanel && not_editing";
+        }
+      ];
+      description = ''
+        Configuration written to {file}`$HOME/.config/zed/keymap.json`.
+        Please reference [zed's documentation] for config options.
+
+        [zed's documentation]: https://zed.dev/docs/key-bindings
+      '';
+    };
+
+    snippets = mkOption {
+      type = attrsOf json.type;
+      default = {};
+      example = literalExample ''
+        {
+          snippets = {
+            # Each snippet must have a name and body, but the prefix and description are optional.
+            # The prefix is used to trigger the snippet, but when omitted then the name is used.
+            # Use placeholders like $1, $2 or ''${1:defaultValue} to define tab stops.
+            # The $0 determines the final cursor position.
+            # Placeholders with the same value are linked.
+            "Log to console" = {
+              prefix = "log";
+              body = ["console.info(\"Hello, ''${1:World}!\")" "$0"];
+              description = "Logs to console";
+            };
+          };
+        }
+      '';
+      description = ''
+        Custom scoped snippets written to {file}`$HOME/.config/zed/snippets/*.json`
+        Please reference [zed's documentation] for more details.
+
+        [zed's documentation]: https://zed.dev/docs/snippets
+      '';
+    };
+
+    themes = mkOption {
+      type = attrsOf json.type;
+      default = {};
+      description = ''
+        Custom themes written to {file}`$HOME/.config/zed/themes/*.json`
+        Please reference [zed's documentation] for more details.
+
+        [zed's documentation]: https://zed.dev/docs/extensions/themes
+      '';
+    };
+
+    tasks = mkOption {
+      type = json.type;
+      default = {};
+      example = [
+        {
+          label = "Example task";
+          command = "for i in {1..5}; do echo \"Hello $i/5\"; sleep 1; done";
+          env = {"foo" = "bar";};
```

Quotation marks aren't necessary in an attrset's name, unless there's special characters that need escaping.

```suggestion
          env = {foo = "bar";};
```